src/main/webapp/script.js

```diff
@@ -143,18 +143,78 @@ function unfreezePage() {
  * Updates the page with new uploaded image.
  * It calls blurring method, puts new and blurred
  * images on canvases, updates blurRadius input
- * bar for new image and download button for
- * every new blurred image.
+ * bar for new image, download button for
+ * every new blurred image and adds click
+ * event on output canvas for specific blur.
  * @param {ImageObject} image
  */
 async function processImage(image) {
+  updateBlurRadiusInputBar(image);
+
+  updateBlurredImage(image);
+
   const outputCanvas = document.getElementById('output-canvas');
+  outputCanvas.onclick = (event) => {
+    toggleAreaBlur(event, image);
+  };
+}
 
-  updateBlurRadiusInputBar(image);
+/**
+ * Function to handle click on output canvas
+ * and toggle blurness of rects to blur
+ * if any were clicked.
+ * @param {MouseEvent} event
+ * @param {ImageObject} image
+ */
+function toggleAreaBlur(event, image) {
+  const outputCanvas = document.getElementById('output-canvas');
 
+  // coordinates of click.
+  const eventX = event.offsetX;
+  const eventY = event.offsetY;
+
+  let wasSomeRectClicked = false;
+
+  for (const blurRect of image.blurAreas) {
+    // Scale coordinate from image coordinates to canvas coordinates.
+    const scaleForCanvas = (coordinate) => {
+      return coordinate * outputCanvas.clientWidth / image.object.width;
+    };
+
+    // blurRect has all properties in original image coordinates,
+    // click event is in canvas coordinates.
+    // Scale blurRect to be in canvas coordinates.
+    const scaledRect = {
+      'leftX': scaleForCanvas(blurRect.leftX),
+      'topY': scaleForCanvas(blurRect.topY),
+      'rightX': scaleForCanvas(blurRect.leftX +
+          blurRect.width),
+      'bottomY': scaleForCanvas(blurRect.topY +
+          blurRect.height),
+    };
+
+    if (scaledRect.leftX <= eventX && eventX <= scaledRect.rightX &&
+        scaledRect.topY <= eventY && eventY <= scaledRect.bottomY) {
```

Comment:
If I read it correctly, the codition should never hold for Y. Could you double check it? Ah, I see now. In the lines above, when you construct scaledRect, your bottomY > topY (which I think is incorrect).

Reply:
topY is the Y coordinate of top left corner of an image. Is is less than bottom coordinate because the coordinates in canvas start in top left corner, and Y axis is directed down.